mandatory/3-playing-computer.js

```diff
@@ -7,12 +7,18 @@
   Answer the following questions:
 
   1. This program throws an error. Why? (If you can't find it, try executing it).
+  // b has not been defined
   2. Remove the line that throws the error.
   3. What is printed to the console?
+  // 2, 6, 4, 9, 6, 13, 8
   4. How many times is "f1" called?
+  // 2 
   5. How many times is "f2" called?
+  // 3
   6. What value does the "a" parameter take in the first "f1" call?
+  // 8
   7. What is the value of the "a" outer variable when "f1" is called for the first time?
+  // 6
```

Comment:
`a` is originally assigned a value of 6.
On the first iteration of the for loop:
- 1 is added to `a` so its new value is 7
- `f1` is not called because the if condition on line 42 is false
On the second iteration of the for loop:
- 1 is added to `a` so its new value is 8
- `f1` is called because the if condition on line 42 is true, therefore `f1` is called with the value of `a` set to 8